sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -908,6 +915,215 @@ class RotatingCertificateCredential implements TokenCredential {
+}
+
+main().catch((err) => {
+  console.log("error code: ", err.code);
+  console.log("error message: ", err.message);
+  console.log("error stack: ", err.stack);
+});
+```
+
+#### Silent authentication with Authentication Record and Token Cache Persistence Options
+
+Once an application has configured a credential to persist token data and an `AuthenticationRecord`, it is possible to silently authenticate. This example demonstrates an application setting the `tokenCachePersistenceOptions` and retrieving an `AuthenticationRecord` from the local file system to create an `InteractiveBrowserCredential` capable of silent authentication.
+
+```ts
+import {
+  useIdentityExtension,
+  InteractiveBrowserCredential,
+  AuthenticationRecord,
+  deserializeAuthenticationRecord
+} from "@azure/identity";
+import { cachePersistenceExtension } from "@azure/identity-cache-persistence";
+import fs from "fs";
+import { promisify } from "util";
+import path from "path";
+
+useIdentityExtension(cachePersistenceExtension);
+
+export async function main(): Promise<void> {
+  const AUTH_RECORD_PATH = "./tokencache.bin";
+  const readFileAsync = promisify(fs.readFile);
+  const fileContent = await readFileAsync(path.join(process.cwd(), AUTH_RECORD_PATH), {
+    encoding: "utf-8"
+  });
+  const authRecord: AuthenticationRecord = deserializeAuthenticationRecord(fileContent);
+
+  const credential = new InteractiveBrowserCredential({
+    tokenCachePersistenceOptions: {
+      enabled: true
+    },
+    authenticationRecord: authRecord
+  });
+}
+
+main().catch((err) => {
+  console.log("error code: ", err.code);
+  console.log("error message: ", err.message);
+  console.log("error stack: ", err.stack);
+});
+```
+
+The credential created in this example will silently authenticate given that a valid token for corresponding to the `AuthenticationRecord` still exists in the persisted token data. There are some cases where interaction will still be required such as on token expiry, or when additional authentication is required for a particular resource.
+
+### Persisting credentials by configuring TokenCachePersistenceOptions
+
+Many credential implementations in the `@azure/identity` library have an underlying token cache which persists sensitive authentication data such as account information, access tokens, and refresh tokens. By default this data exists in an in memory cache which is specific to the credential instance. However, there are scenarios where an application needs persist it across executions in order to share the token cache across credentials. To accomplish this the `@azure/identity` provides the `tokenCachePersistenceOptions`.
+
+> IMPORTANT! The token cache contains sensitive data and MUST be protected to prevent compromising accounts. All application decisions regarding the persistence of the token cache must consider that a breach of its content will fully compromise all the accounts it contains.
+
+#### Using the default token cache
```

Comment:
This is just the same sample as above when we introduce the concept of token caching for the first time, no? I don't necessarily think we need to repeat it, and it should be enough just to show how to use a named cache and explain the consequence of naming.

Reply:
merged with above sample